src/main/java/io/stargate/sgv2/jsonapi/config/constants/VectorConstant.java

```diff
@@ -1,14 +1,30 @@
 package io.stargate.sgv2.jsonapi.config.constants;
 
 import io.smallrye.config.ConfigMapping;
-import java.util.Set;
+import io.stargate.sgv2.jsonapi.service.schema.SimilarityFunction;
+import java.util.Map;
 
 @ConfigMapping(prefix = "stargate.jsonapi.vector")
 public interface VectorConstant {
   /*
-  Supported Source Models for Vector Index in Cassandra
+  Supported Source Models and suggested function for Vector Index in Cassandra
    */
-  Set<String> SUPPORTED_SOURCES =
-      Set.of(
-          "ada002", "openai_v3_small", "openai_v3_large", "bert", "gecko", "nv_qa_4", "cohere_v3");
+  Map<String, SimilarityFunction> SUPPORTED_SOURCES =
```

Comment:
Is there a cassandra document saying these models are DOT_PRODUCT recommended?

Reply:
Unfortunately no, it's pulled out from the `datastax/cassandra` [code]([https://github.com/datastax/cassandra/blob/88e5905611a142991a4e717a654351aea622a19[…]g/apache/cassandra/index/sai/disk/vector/VectorSourceModel.java](https://github.com/datastax/cassandra/blob/88e5905611a142991a4e717a654351aea622a197/src/java/org/apache/cassandra/index/sai/disk/vector/VectorSourceModel.java#L61)).

Reply:
cool